ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -580,7 +590,14 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{app
                                                 {title: 'Applications', path: '/applications'},
                                                 {title: <ApplicationsDetailsAppDropdown appName={this.props.match.params.name} />}
                                             ],
-                                            actionMenu: {items: this.getApplicationActionMenu(application, true)},
+                                            actionMenu: {
+                                                items: [
+                                                    ...this.getApplicationActionMenu(application, true),
+                                                    ...(this.state.topBarActionMenuExts
+                                                        ?.filter(ext => ext?.shouldDisplay && ext.shouldDisplay(application))
```

Comment:
```suggestion
                                                        ?.filter(ext => ext.shouldDisplay?.(application))
```
I believe you can further simplify this
Also don't think the optional chaining on `ext?.` is necessary since an item would have to exist for it to be iterated over by `filter`